ARCHITECTURE.md

```diff
@@ -0,0 +1,56 @@
+
+1. **High Performance:** Presto strives to exceed data warehouse performance in data lakes. To achieve this, Presto works to achieve vertical integration over modular portions of the stack.
+2. **Scalability:** Presto is used at extremely large data lakes across the world. As such, it aims for high degrees of scalability.
+3. **Connectivity and Integration:** Presto is designed to be flexible and adaptable to changes in infrastructure. This includes a highly customizable plugin infrastructure, including pluggable storage connectors.
+4. **User-Friendly:** Presto strives to be simple enough for a user to spin up and quickly test out, using familiar ANSI style SQL syntax, sensible default values and an optimizer that produces efficient query plans.
+
+### For developers
+
+Presto aims to accomplish the above goals for users by means of creating a broad, powerful, and collaborative open source community that strives for high standards in database engineering and design.
+
+1. **Open Source:** Develop and maintain as an open-source project, encouraging community contributions and collaboration.
+2. **Broad and open community:** Foster a large and active community to drive the project’s adoption and direction.
+3. **Composability**: Presto is designed for lakes and lakehouses. As such, it is engineered to fit cohesively in an ecosystem of other tools, and strives to integrate with industry standard specifications and libraries that help accomplish the user goals above.
+4. **Coding excellence**: Presto strives to maintain a high bar for contributions through good design, good abstractions, rigorous tests, and documentation.
+
+## Architecture
+
+Presto follows a distributed system model with a coordinator and multiple worker nodes. [See Presto Concepts for more information.](https://prestodb.io/docs/0.285.1/overview/concepts.html#server-types)
+
+Long term initiatives in Presto [have corresponding project boards](https://github.com/prestodb/presto/projects?query=is%3Aopen). Current progress can be roughly understood by looking at the project boards.
+
+## Long Term Vision
+
+### Vertical integration in data lakes
+
+Presto aims to be the top performing system for data lakes. The top priority for the project is to move fully onto a native evaluation engine, particularly [Velox](https://velox-lib.io/), with the same expectations outlined in the vision for users, emphasizing user-friendliness and connectivity.
+
+Motivations for this effort are numerous:
+
+1. It is expected to be too difficult to achieve data warehouse performance using a Java evaluation engine. Some optimizations, such as vectorized execution in particular, require flexibility and precision which is difficult to achieve in Java.
```

Comment:
"to achieve data warehouse performance" isn't really a goal. I think we just want to say we think C++ will be faster